xbmc/epg/EpgContainer.cpp

```diff
@@ -245,16 +245,17 @@ bool CEpgContainer::PersistAll(void)
 {
   bool bReturn(true);
   m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
+  EPGMAP copy = m_epgs;
   m_critSection.unlock();
 
-  for (EPGMAP_CITR it = copy.begin(); it != copy.end() && !m_bStop; it++)
+  for (const auto &epgEntry : copy)
   {
-    CEpg *epg = it->second;
+    CEpg *epg = epgEntry.second;
     if (epg && epg->NeedsSave())
-    {
       bReturn &= epg->Persist();
-    }
+
+    if (m_bStop)
```

Comment:
comes with "auto &it", "const auto &it" and "const auto it" and was:
```
EpgContainer.cpp:251:70: error: passing ‘const std::_Rb_tree_iterator<std::pair<const unsigned int, EPG::CEpg*> >’ as ‘this’ argument of ‘std::_Rb_tree_iterator<_Tp>::_Self& std::_Rb_tree_iterator<_Tp>::operator++() [with _Tp = std::pair<const unsigned int, EPG::CEpg*>; std::_Rb_tree_iterator<_Tp>::_Self = std::_Rb_tree_iterator<std::pair<const unsigned int, EPG::CEpg*> >]’ discards qualifiers [-fpermissive]
   for (const auto it = copy.begin(); it != copy.end() && !m_bStop; ++it)
```

Randomly I has added this and has worked:
```C++
for (auto it = copy.begin(); it != copy.end() && !m_bStop; ++it)
```